local_pathfinding/ompl_path.py

```diff
@@ -218,3 +177,44 @@ def is_state_valid(state: ob.SE2StateSpace) -> bool:
     # TODO: implement obstacle avoidance here
     # note: `state` is of type `SE2StateInternal`, so we don't need to use the `()` operator.
     return state.getX() < 0.6
+
+
+def choose_planner(planner: str, si: ob.SpaceInformation) -> ob.Planner:
+    """Choose the planner to use for the OMPL query.
+
+    Args:
+        planner (str): Name of the planner to use.
+        si (ob.SpaceInformation): Encapsulates the planning problem to be solved.
+
+    Returns:
+        ob.Planner: Planner to use for the OMPL query.
+    """
+    match planner.lower():
+        case "bitstar":
+            return og.BITstar(si)
+        case "bfmtstar":
+            return og.BFMT(si)
+        case "fmtstar":
+            return og.FMT(si)
+        case "informedrrtstar":
+            return og.InformedRRTstar(si)
+        case "lazylbtrrt":
+            return og.LazyLBTRRT(si)
+        case "lazyprmstar":
+            return og.LazyPRMstar(si)
+        case "lbtrrt":
+            return og.LBTRRT(si)
+        case "prmstar":
+            return og.PRMstar(si)
+        case "rrtconnect":
+            return og.RRTConnect(si)
+        case "rrtsharp":
+            return og.RRTsharp(si)
+        case "rrtstar":
+            return og.RRTstar(si)
+        case "rrtxstatic":
+            return og.RRTXstatic(si)
+        case "sorrtstar":
+            return og.SORRTstar(si)
+        case _:
+            return og.RRTStar(si)
```

Comment:
I'm still not completely satisfied with this. Ideally I would want data validation of the planner ros param to occur in `get_params()` node_navigate.py (error log if not valid) but for it to be converted into `og.Planner` here. If you have any ideas feel free to suggest/implement them, I'll keep thinking about this as well

Reply:
I attempted to tackle this in my latest commit, what do you think?